kernel/src/expressions/mod.rs

```diff
@@ -250,6 +251,22 @@ pub struct OpaquePredicate {
     pub op: OpaquePredicateOpRef,
     pub exprs: Vec<Expression>,
 }
+fn fail_serialize_opaque_predicate<S>(
+    _value: &OpaquePredicate,
+    _serializer: S,
+) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    Err(ser::Error::custom("Cannot serialize Opaque Expression"))
```

Comment:
why not force opaque predicate/expression to be serializable? 

Is is possible OpaquePredicate/Expression have a debug string that could be added for context in the error?

Reply:
I guess this is the same comment made above by @scovich